src/main.py

```diff
@@ -185,33 +186,28 @@ def get_friends(self):
         self.friends = discord_integration.get_friends()
 
         for friend in self.friends:
-            user = friend["user"]
+            user = friend.user
 
-            button = QPushButton(
-                text=friend["nickname"] if friend["nickname"] else user["global_name"]
-            )
+            button = QPushButton(text=friend.get_friend_name())
 
             def set_button_icon(button, data):
                 pixmap = QPixmap()
                 if pixmap.loadFromData(data):
                     button.setIcon(QIcon(pixmap))
-                else:
-                    fallback_path = os.path.join(current_dir, "assets", "user.png")
-                    button.setIcon(QIcon(fallback_path))
-
-            avatar_url = f"https://cdn.discordapp.com/avatars/{user['id']}/{user['avatar']}.webp?size=128"
-            worker = FileRequestWorker(avatar_url, "users")
-            worker.finished.connect(
-                (lambda button: lambda data: set_button_icon(button, data))(button)
-            )
-            self.threadpool.start(worker)
+
+            if avatar_url := user.get_avatar_url():
```

Comment:
Does `:=` mean that it's assigning `avatar_url` to `user.get_avatar_url()` and comparing it at the same time?

Reply:
That's the usually known as the "walrus operator". It isn't exactly making a comparison, it assigns the variable `avatar_url` to `user.get_avatar_url()` and then evaluates the variable as an expression. I use it here to avoid the lengthier equivalent:

```
avatar_url = user.get_avatar_url()
if avatar_url:
   ... 
```

Or even:


```
if user.get_avatar_url():
   avatar_url = user.get_avatar_url()
   ... 
```

Reply:
Got it. Still a bit weird to read.